include/linux/virtio_shm.h

```diff
@@ -10,6 +10,8 @@
 #ifdef CONFIG_VIRTIO_IVSHMEM
 struct page *virtio_shmem_allocate_page(struct device *dev);
 void virtio_shmem_free_page(struct device *dev, struct page *page);
+void *virtio_shmem_alloc(struct device *dev, size_t size);
+void virtio_shmem_free(struct device *dev, void *addr, size_t size);
```

Comment:
there already has function virtio_shmem_allocate_page to alloc, why adding one more virtio_shmem_alloc?

Reply:
This provide virtual address, virtio_shmem_allocate_page provide page pointer for dma.